scripts/redis/scan_redis.py

```diff
@@ -60,10 +72,27 @@ def run():
         for key in data:
             key_type = client.type(key)
             size = get_size(client, key, key_type)
+
+            # ttl() returns None in redis 2.x and -1 in redis 3.x for
+            # keys that don't have an expiration.
             ttl = client.ttl(key)
-            log_file.write('%s %s %s %d\n' % (key, key_type, str(ttl), size))
+
+            new_ttl = None
+            if set_ttl == -1:
+                client.persist(key)
+                new_ttl = -1
+            elif set_ttl is not None and (ttl == -1 or ttl is None):
+                # Only change TTLs for keys with no TTL
+                client.expire(key, set_ttl)
+                new_ttl = set_ttl
+
+            line = '%s %s %s %s %d' % (key, key_type, ttl, new_ttl, size,)
```

Comment:
Dump a JSON dict per line instead. Easier to process with `jq` later if needed.

Reply:
Maybe a future improvement. Right now I am use to just using `awk`, `uniq`, `grep` to analyze the output.